pageserver/src/tenant/mgr.rs

```diff
@@ -1539,14 +1539,7 @@ impl SlotGuard {
     /// is responsible for protecting
     fn old_value_is_shutdown(&self) -> bool {
         match self.old_value.as_ref() {
-            Some(TenantSlot::Attached(tenant)) => {
-                // TODO: PR #5711 will add a gate that enables properly checking that
-                // shutdown completed.
-                matches!(
-                    tenant.current_state(),
-                    TenantState::Stopping { .. } | TenantState::Broken { .. }
-                )
-            }
```

Comment:
Is all of the TenantState usage now gone in general, or was the gate closed at the end of the Tenant::shutdown (and on broken transition)?

Reply:
No, this particular use of TenantState was only added in the previous PR as a placeholder: there are still other places where it is used, notably in the tenant deletion after restart flow, where a Tenant goes into Stopping state immediately after being loaded.

Reply:
Please do not resolve this yet.

Reply:
Ok, Tenant::shutdown and Timeline::shutdown first cancel the token, then close the gates. This looks good to me.